Infrastructure-Scripts/Get-EnvironmentDeploymentRecords.ps1

```diff
@@ -0,0 +1,62 @@
+<#
+    .SYNOPSIS
+    Gets the latest deployment record for a given pipeline and environment and checks whether the current instance is able to run without clashing with said record.
+
+    .DESCRIPTION
+    Gets payload from Azure DevOps RestAPI and filters it to produce the build ID of the newest running build. Then compares to its own build ID, if equal to it then run and if not sleep and redo.
+
+    .PARAMETER EnvironmentId
+    The ID number associated with the environemnt stage being used for the deployment,
+
+    .PARAMETER PipelineName
+    The name of the pipeline that the deployment is being run on.
+
+    .PARAMETER RunId
+    The Run ID of the current deployment which is used to compare to the lowest active RunID returned by the API
+
+    .PARAMETER SleepTime
+    The the amount of time in seconds that the process will wait before retrying the comparison if there is another deployment currently running.
+
+    .EXAMPLE
+    .\Get-EnvironmentDeploymentRecords.ps1 -EnvironmentId 139 -PipelineName das-levy-transfer-matching-api -RunId 460XXX
+#>
+
+[CmdletBinding()]
+Param (
+    [Parameter(Mandatory = $true)]
+    [String]$EnvironmentId,
+    [Parameter(Mandatory = $true)]
+    [String]$PipelineName,
+    [Parameter(Mandatory = $true)]
+    [Int]$RunId,
+    [Parameter(Mandatory = $false)]
+    [Int]$SleepTime = 20
+)
+
+$Url = "$env:SYSTEM_ORGANISATIONNAME/$env:SYSTEM_PROJECTNAME/_apis/distributedtask/environments/$EnvironmentId/environmentdeploymentrecords?top=100?api-version=6.0-preview.1"
+$RetryCounter = 1
+
+while ($RetryCounter -lt 10){
+    try{
+        Write-Verbose "Attempt $RetryCounter"
+        #Invoke call to Azure DevOps Rest API to get all build data for given environment.
+        $EnvironmentPipelineRuns = Invoke-RestMethod -Method GET -Uri $Url -Headers  @{ Authorization = "Bearer $env:SYSTEM_ACCESSTOKEN" } -TimeoutSec 30
+    }
+    catch{
+        Write-Verbose "Response code $($_.Exception.Response.StatusCode.Value__) received"
```

Comment:
Change to `Write-Error` and add `.  Terminating process, deployment will continue` to the end of the message